### PR TITLE
Switch to pipenv and Pipfile

### PR DESCRIPTION
**Pull request by yukinchan**

### CODE REVIEW

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -38,9 +30,8 @@ do
             BEAT_MODE=true
             ;;
         -h | --help )
-            echo "usage: setup.sh [-h] [--venv VENV_PATH]
+            echo "usage: setup.sh [-h]
```
Comment (waynesun09) [inline docker/entrypoint.sh]: Need update with setup.sh which is removed now

Comment (yukinchan) [reply docker/entrypoint.sh]: Fixed in b16a87b.

**Comment by waynesun09** [comment]
@YukinChan pls rebase and update the PR, thanks

**Comment by waynesun09** [comment]
@YukinChan Please rebase as repo udpated and squash commits if necessary

**Comment by yukinchan** [comment]
> @YukinChan Please rebase as repo udpated and squash commits if necessary

Rebased and updated.

**Comment by ryncsn** [comment]
Thanks, looks great. Didn't apply this because previously openshift online still require that requirements.txt to work, now it should work just fine with Pipenv file.

**Comment by yukinchan** [comment]
> Thanks, looks great. Didn't apply this because previously openshift online still require that requirements.txt to work, now it should work just fine with Pipenv file.

I just found an issue in `Dockerfile` that `ENTRYPOINT` is wrongly set now because the `docker/` folder is now `depoly`.

**Comment by ryncsn** [comment]
> > Thanks, looks great. Didn't apply this because previously openshift online still require that requirements.txt to work, now it should work just fine with Pipenv file.
> 
> I just found an issue in `Dockerfile` that `ENTRYPOINT` is wrongly set now because the `docker/` folder is now `depoly`.

Thanks for pointing it out, I fixed it together with the CI test, no longer a problem.

**Comment by yukinchan** [comment]
> > > Thanks, looks great. Didn't apply this because previously openshift online still require that requirements.txt to work, now it should work just fine with Pipenv file.
> > 
> > 
> > I just found an issue in `Dockerfile` that `ENTRYPOINT` is wrongly set now because the `docker/` folder is now `depoly`.
> 
> Thanks for pointing it out, I fixed it together with the CI test, no longer a problem.

That's very nice of you for fixing this. Have to take some time to see your recent changes and merge them into production branch :)

**Pull request merged**